assets/js/atomic/utils/index.js

```diff
@@ -0,0 +1,193 @@
+import {
+	ProductListLink,
+	ProductListTitle,
+	ProductListPrice,
+	ProductListButton,
+	ProductListImage,
+	ProductListRating,
+	ProductListSummary,
+	ProductListSaleBadge,
+} from '../components/product-list';
+
+/**
+ * List of mapped components.
+ */
+export const COMPONENT_MAP = {
+	ProductListLink,
+	ProductListTitle,
+	ProductListPrice,
+	ProductListButton,
+	ProductListImage,
+	ProductListRating,
+	ProductListSummary,
+	ProductListSaleBadge,
+};
+
+// @todo how to support these using mapping?
+//'core/paragraph',
+//'core/heading',
+/**
+ * Map blocks names to component names.
+ */
+export const BLOCK_MAP = [
+	{
+		component: 'ProductListPrice',
+		blockName: 'woocommerce/product-list-price',
+	},
+	{
+		component: 'ProductListImage',
+		blockName: 'woocommerce/product-list-image',
+	},
+	{
+		component: 'ProductListTitle',
+		blockName: 'woocommerce/product-list-title',
+	},
+	{
+		component: 'ProductListRating',
+		blockName: 'woocommerce/product-list-rating',
+	},
+	{
+		component: 'ProductListButton',
+		blockName: 'woocommerce/product-list-button',
+	},
+	{
+		component: 'ProductListSummary',
+		blockName: 'woocommerce/product-list-summary',
+	},
+	{
+		component: 'ProductListSaleBadge',
+		blockName: 'woocommerce/product-list-sale-badge',
+	},
+];
+
+/**
+ * The default layout built from the default template.
+ */
+export const DEFAULT_PRODUCT_LIST_LAYOUT = [
+	{
+		component: 'ProductListSaleBadge',
+		props: { align: 'left' },
+	},
+	{
+		component: 'ProductListImage',
+		props: {},
+	},
+	{
+		component: 'ProductListTitle',
+		props: {},
+	},
+	{
+		component: 'ProductListPrice',
+		props: {},
+	},
+	{
+		component: 'ProductListRating',
+		props: {},
+	},
+	{
+		component: 'ProductListButton',
+		props: {},
+	},
+];
+
+/**
+ * Converts and maps a layoutConfig to a block template.
+ */
+export const layoutConfigToBlockTemplate = ( layoutConfig ) => {
+	const templateConfig = [];
+	const blockLookup = BLOCK_MAP.reduce(
+		( acc, it ) => ( ( acc[ it.component ] = it ), acc ),
+		{}
+	);
+
+	layoutConfig.map( ( layout ) => {
+		const block = blockLookup[ layout.component ] || null;
+
+		if ( block ) {
+			templateConfig.push( [ block.blockName, layout.props ] );
+		}
+
+		return true;
+	} );
+
+	return templateConfig;
+};
+
+/**
+ * The default template (list of inner blocks) for the product list.
+ */
+export const DEFAULT_PRODUCT_LIST_TEMPLATE = layoutConfigToBlockTemplate(
+	DEFAULT_PRODUCT_LIST_LAYOUT
+);
+
+/**
+ * Converts innerblocks to a list of layout configs.
+ *
+ * @param {object} innerBlocks Inner block components.
```

Comment:
`innerBlocks` is an array of objects, no?
```suggestion
 * @param {object[]} innerBlocks Inner block components.
```